### PR TITLE
Setup router & upload-function

### PR DESCRIPTION
**Pull request by mariothomsen**
- Added an upload-function
- Added a router-function 

Issue: https://github.com/mariothomsen/keepit/issues/1
Preview: https://keepit-6wm1a32jh.vercel.app/

### CODE REVIEW

File: frontend/src/Components/UploadButton.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <div>
+      <ImageUploading
+        multiple
+        onChange={onChange}
+        maxNumber={maxNumber}
+        dataURLKey="data_url"
+      >
+        {({ onImageUpload }) => (
+          <div>
+            <button onClick={onImageUpload}> NEW KEEPit</button>
```
Comment (lmachens) [inline frontend/src/Components/UploadButton.js:27]:
```suggestion
            <button onClick={onImageUpload}>NEW KEEPit</button>
```

File: frontend/src/Components/UploadButton.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  const onChange = (imageList, addUpdateIndex) => {
+    history.push('/new', { images: imageList })
+    if (location.pathname === '/new') {
+      window.location.reload()
+    }
+  }
+
+  return (
+    <div>
```
Comment (lmachens) [inline frontend/src/Components/UploadButton.js:18]: I guess you don't need this `div` (or is it required for some styling?) here

File: frontend/src/Pages/NewKeepitPage.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const history = useHistory()
+  const [images, setImages] = useState()
+  const [showUploadButton, setShowUploadButton] = useState(false)
+
+  useEffect(() => {
+    if (images && images.length > 0) {
+      setShowUploadButton(false)
+    } else {
+      setShowUploadButton(true)
+    }
```
Comment (lmachens) [inline frontend/src/Pages/NewKeepitPage.js:15]: You can simplify this:

```suggestion
    setShowUploadButton(images && images.length > 0))   
```

or (more magic going on here 🎉)

```suggestion
    setShowUploadButton(Boolean(images?.length))   
```

File: frontend/src/Pages/NewKeepitPage.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { useHistory } from 'react-router-dom'
+import { useState, useEffect } from 'react'
+import UploadButton from '../Components/UploadButton'
+
+export default function NewKeepitPage() {
+  const history = useHistory()
+  const [images, setImages] = useState()
+  const [showUploadButton, setShowUploadButton] = useState(false)
```
Comment (lmachens) [inline frontend/src/Pages/NewKeepitPage.js:8]: You could remove this state and effect if you do this:

```suggestion
  const showUploadButton = images && images.length > 0;
```

or like in the next comment suggested:

```suggestion
  const showUploadButton = Boolean(images?.length);
```

File: frontend/src/Pages/NewKeepitPage.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  return (
+    <div>
+      New keepit!<br></br>
+      {images &&
+        images.map((image, index) => (
+          <div key={index}>
+            <img src={image['data_url']} alt="" width="100" />
+            <div>
```
Comment (lmachens) [inline frontend/src/Pages/NewKeepitPage.js:39]: I think this `div` is not required. If you added it to create a block element, consider using CSS for this.

File: frontend/src/Pages/NewKeepitPage.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+              <button onClick={() => remove(index)}>Remove</button>
+            </div>
+          </div>
+        ))}
+      {showUploadButton ? <UploadButton></UploadButton> : ''}
+      {!showUploadButton ? (
+        <button onClick={uploadToBe}>Save Keepit</button>
+      ) : (
+        ''
+      )}
```
Comment (lmachens) [inline frontend/src/Pages/NewKeepitPage.js:49]:
```suggestion
      {showUploadButton ? <UploadButton /> : <button onClick={uploadToBe}>Save Keepit</button>}
```

**Review by lmachens — COMMENTED** [review]
Good structured and easy to review. Well done 👍

File: frontend/src/Pages/NewKeepitPage.js (hunk 4 above)
Comment (mariothomsen) [reply frontend/src/Pages/NewKeepitPage.js:8]: Great, thanks!

**Pull request merged**